From e58f91a02598f0c5a2e642b94e6e2112b2662d85 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Wed, 12 Oct 2022 16:38:36 +0200
Subject: [PATCH] ARM: dts: stm32: add A7 examples dts files for stm32mp157

NOT_UPSTREAMABLE

This commit adds new files to be able to run Linux examples on MP15
EV1 and DKx boards.

Change-Id: Ia7589bbdf3e63cc8c498befe43ca4f37a37565bc
Signed-off-by: Alexandre Torgue <alexandre.torgue@foss.st.com>
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
---
 arch/arm/boot/dts/st/Makefile                     | 10 +++++++++-
 .../boot/dts/st/stm32mp157a-dk1-a7-examples.dts   | 14 ++++++++++++++
 .../boot/dts/st/stm32mp157a-ev1-a7-examples.dts   | 15 +++++++++++++++
 .../boot/dts/st/stm32mp157c-dk2-a7-examples.dts   | 14 ++++++++++++++
 .../boot/dts/st/stm32mp157c-ev1-a7-examples.dts   | 15 +++++++++++++++
 .../boot/dts/st/stm32mp157d-dk1-a7-examples.dts   | 14 ++++++++++++++
 .../boot/dts/st/stm32mp157d-ev1-a7-examples.dts   | 15 +++++++++++++++
 .../boot/dts/st/stm32mp157f-dk2-a7-examples.dts   | 14 ++++++++++++++
 .../boot/dts/st/stm32mp157f-ev1-a7-examples.dts   | 15 +++++++++++++++
 9 files changed, 125 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/boot/dts/st/stm32mp157a-dk1-a7-examples.dts
 create mode 100644 arch/arm/boot/dts/st/stm32mp157a-ev1-a7-examples.dts
 create mode 100644 arch/arm/boot/dts/st/stm32mp157c-dk2-a7-examples.dts
 create mode 100644 arch/arm/boot/dts/st/stm32mp157c-ev1-a7-examples.dts
 create mode 100644 arch/arm/boot/dts/st/stm32mp157d-dk1-a7-examples.dts
 create mode 100644 arch/arm/boot/dts/st/stm32mp157d-ev1-a7-examples.dts
 create mode 100644 arch/arm/boot/dts/st/stm32mp157f-dk2-a7-examples.dts
 create mode 100644 arch/arm/boot/dts/st/stm32mp157f-ev1-a7-examples.dts

--- a/arch/arm/boot/dts/st/Makefile
+++ b/arch/arm/boot/dts/st/Makefile
@@ -38,8 +38,10 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157a-avenger96.dtb \
 	stm32mp157a-dhcor-avenger96.dtb \
 	stm32mp157a-dk1.dtb \
+	stm32mp157a-dk1-a7-examples.dtb \
 	stm32mp157a-ed1.dtb \
 	stm32mp157a-ev1.dtb \
+	stm32mp157a-ev1-a7-examples.dtb \
 	stm32mp157a-iot-box.dtb \
 	stm32mp157a-microgea-stm32mp1-microdev2.0.dtb \
 	stm32mp157a-microgea-stm32mp1-microdev2.0-of7.dtb \
@@ -50,20 +52,26 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157c-dhcom-pdk2.dtb \
 	stm32mp157c-dhcom-picoitx.dtb \
 	stm32mp157c-dk2.dtb \
+	stm32mp157c-dk2-a7-examples.dtb \
 	stm32mp157c-ed1.dtb \
 	stm32mp157c-emsbc-argon.dtb \
 	stm32mp157c-ev1.dtb \
+	stm32mp157c-ev1-a7-examples.dtb \
 	stm32mp157c-lxa-mc1.dtb \
 	stm32mp157c-lxa-tac-gen1.dtb \
 	stm32mp157c-lxa-tac-gen2.dtb \
 	stm32mp157c-odyssey.dtb \
 	stm32mp157c-phycore-stm32mp1-3.dtb \
 	stm32mp157d-dk1.dtb \
+	stm32mp157d-dk1-a7-examples.dtb \
 	stm32mp157d-ed1.dtb \
 	stm32mp157d-ev1.dtb \
+	stm32mp157d-ev1-a7-examples.dtb \
 	stm32mp157f-dk2.dtb \
+	stm32mp157f-dk2-a7-examples.dtb \
 	stm32mp157f-ed1.dtb \
-	stm32mp157f-ev1.dtb
+	stm32mp157f-ev1.dtb \
+	stm32mp157f-ev1-a7-examples.dtb
 dtb-$(CONFIG_ARCH_U8500) += \
 	ste-snowball.dtb \
 	ste-hrefprev60-stuib.dtb \
--- /dev/null
+++ b/arch/arm/boot/dts/st/stm32mp157a-dk1-a7-examples.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157a-dk1.dts"
+
+/ {
+	model = "STMicroelectronics STM32MP157A-DK1 configured to run Linux A7 examples";
+	compatible = "st,stm32mp157a-dk1-a7-examples", "st,stm32mp157a-dk1", "st,stm32mp157";
+};
--- /dev/null
+++ b/arch/arm/boot/dts/st/stm32mp157a-ev1-a7-examples.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157a-ev1.dts"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "STMicroelectronics STM32MP157A-EV1 configured to run Linux A7 examples";
+	compatible = "st,stm32mp157a-ev1-a7-examples", "st,stm32mp157a-ev1", "st,stm32mp157a-ed1", "st,stm32mp157";
+};
--- /dev/null
+++ b/arch/arm/boot/dts/st/stm32mp157c-dk2-a7-examples.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157c-dk2.dts"
+
+/ {
+	model = "STMicroelectronics STM32MP157C-DK2 configured to run Linux A7 examples";
+	compatible = "st,stm32mp157c-dk2-a7-examples", "st,stm32mp157c-dk2", "st,stm32mp157";
+};
--- /dev/null
+++ b/arch/arm/boot/dts/st/stm32mp157c-ev1-a7-examples.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157c-ev1.dts"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "STMicroelectronics STM32MP157C-EV1 configured to run Linux A7 examples";
+	compatible = "st,stm32mp157c-ev1-a7-examples", "st,stm32mp157c-ev1", "st,stm32mp157c-ed1", "st,stm32mp157";
+};
--- /dev/null
+++ b/arch/arm/boot/dts/st/stm32mp157d-dk1-a7-examples.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157d-dk1.dts"
+
+/ {
+	model = "STMicroelectronics STM32MP157D-DK1 configured to run Linux A7 examples";
+	compatible = "st,stm32mp157d-dk1-a7-examples", "st,stm32mp157d-dk1", "st,stm32mp157";
+};
--- /dev/null
+++ b/arch/arm/boot/dts/st/stm32mp157d-ev1-a7-examples.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157d-ev1.dts"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "STMicroelectronics STM32MP157C-EV1 configured to run Linux A7 examples";
+	compatible = "st,stm32mp157d-ev1-a7-examples", "st,stm32mp157d-ev1", "st,stm32mp157d-ed1", "st,stm32mp157";
+};
--- /dev/null
+++ b/arch/arm/boot/dts/st/stm32mp157f-dk2-a7-examples.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157f-dk2.dts"
+
+/ {
+	model = "STMicroelectronics STM32MP157F-DK2 configured to run Linux A7 examples";
+	compatible = "st,stm32mp157f-dk2-a7-examples", "st,stm32mp157f-dk2", "st,stm32mp157";
+};
--- /dev/null
+++ b/arch/arm/boot/dts/st/stm32mp157f-ev1-a7-examples.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157f-ev1.dts"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "STMicroelectronics STM32MP157F-EV1 configured to run Linux A7 examples";
+	compatible = "st,stm32mp157f-ev1-a7-examples", "st,stm32mp157f-ev1", "st,stm32mp157f-ed1", "st,stm32mp157";
+};
